src/data-sources/abstract.dapp.data-source.ts

```diff
@@ -0,0 +1,13 @@
+import { AbstractDataSource, Meta } from './abstract.data-source';
+import { Protocol } from '../entities/protocol.entity';
+
+export abstract class AbstractDappDataSource<C, P> extends AbstractDataSource<
```

Comment:
since we do not have any new entities, why do we create new abstract class here?